### ISSUE TITLE
Expected type warning for `list*`

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#610]
This doesn't trigger a type error:

```
(list* 1 2 3 'foo)
```

Note:

```
$ clj -A:clj-kondo --lint - <<< '(seq (symbol "foo"))'
linting took 51ms, errors: 0, warnings: 0
```

**Comment by borkdude** [comment]
TODO, write tests:

```
$ clj -A:clj-kondo --lint - <<< '(seq (symbol "foo"))'
<stdin>:1:6: warning: Expected: seqable collection, received: symbol.
linting took 57ms, errors: 0, warnings: 1
```
